ENIGMAsystem/SHELL/Universal_System/bufferstruct.cpp

```diff
@@ -308,13 +308,11 @@ variant buffer_peek(int buffer, unsigned offset, int type) {
   binbuff->Seek(offset);
   if (type != buffer_string) {
     //unsigned dsize = buffer_sizeof(type) + binbuff->alignment - 1;
-    unsigned char data[buffer_sizeof(type)];
     //NOTE: These buffers most likely need a little more code added to take care of endianess on different architectures.
     //TODO: Fix floating point precision.
     long res = 0;
     for (unsigned i = 0; i < buffer_sizeof(type); i++) {
-      data[i] = binbuff->ReadByte();
-      res += data[i] << i * 8;
+      res += binbuff->ReadByte() << i * 8;
```

Comment:
@RobertBColton did this too idk

Reply:
@JoshDreamland I added the `buffer_*` functions, and they never worked. They might work a little bit, but they totally need a do-over. I last attempted this API in like 2013, so if I tried it now, it might come out a lot better. Regardless, I can't really tell you why I had that unused buffer there to begin with.
https://docs2.yoyogames.com/source/_build/3_scripting/4_gml_reference/buffers/index.html